src/bezier.cpp

```diff
@@ -191,21 +198,15 @@ double Curve::iterateByLength(Parameter t, double s, double epsilon, std::size_t
   if (s_t + s > length())
     return 1;
 
-  std::size_t current_iter = 0;
-  while (current_iter < max_iter)
+  double f = (length(t) - s_t - s);
```

Comment:
s_t is equal to length(t), isn't it? -s is sufficient for the initialization.

As I can see, you completely removed max_iter (probably because of consistency), but is there any change for the loop to diverge?